File: frontend/src/api/notification-msg.ts
import request, { type PageResult } from './request'

export interface NotificationMsg {
  id: number
  type: string // build_complete / deploy_complete / deploy_failed / announcement
  title: string
  content: string
  level: string // info / success / warning / error
  read: boolean
  pipelineId: number
  createdAt: string
}

export const listNotificationMsgs = (page = 1, pageSize = 20) =>
  request.get<any, PageResult<NotificationMsg>>('/notification-msgs', { params: { page, pageSize } })

export const getUnreadCount = () =>
  request.get<any, { count: number }>('/notification-msgs/unread-count')

export const markRead = (id: number) =>
  request.put<any, null>(`/notification-msgs/${id}/read`)

export const markAllRead = () =>
  request.put<any, null>('/notification-msgs/read-all')

export const publishAnnouncement = (title: string, content: string) =>
  request.post<any, null>('/notification-msgs/announce', { title, content })

File: backend/internal/model/notification_msg.go
package model

import "time"

// NotificationMsg represents a system notification message
type NotificationMsg struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	Type      string    `gorm:"size:32;not null;index" json:"type"` // build_complete / deploy_complete / deploy_failed / announcement
	Title     string    `gorm:"size:256;not null" json:"title"`
	Content   string    `gorm:"type:text" json:"content"`
	Level     string    `gorm:"size:20;default:info" json:"level"` // info / success / warning / error
	Read      bool      `gorm:"default:false" json:"read"`
	PipelineID uint     `gorm:"index" json:"pipelineId"`
	CreatedAt time.Time `json:"createdAt"`
}
